### PR TITLE
[4975376][5541172]perplexity and kl-divergence benchmark metrics

### PR DESCRIPTION
**Pull request by ynankani**
## What does this PR do?

**Type of change:** new tests, accuracy benchmark metrics

**Overview:**  Added perplexity and kl-divergence metrics as part of accuracy benchmark

## Usage

```python
python compute_kl_divergence.py --hf_model "F:\shared\Llama-3.1-8B-Instruct" --ep cuda --path "G:\llama3.1-8B-genai-cuda-fp16" --ep directml --path "G:\shared\llama3.1-8B-genai-dml_fp16"  --output "comparison_results.json"

python run_perplexity.py --models  G:\llama3.1-8B-genai-cuda-fp16 --i 1024,2048,4096,8192,12288 --chunk_size=1024
```

## Testing
Tested and collected scores for below models

1. Llama-3.1-8B-Instruct
2. DeepSeek-R1-Distill-Qwen-7B
3. Qwen2.5-1.5B-Instruct

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: NA
- **Did you write any new necessary tests?**: NA
- **Did you add or update any necessary documentation?**: Yes
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: NA

### CODE REVIEW

File: examples/windows/accuracy_benchmark/perplexity_metrics/perplexity_metrics.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+                valid_indices = np.arange(0, trg_len - 1)
+                valid_targets = target_ids_shifted[-trg_len + 1 :]
+            else:
+                valid_indices = np.arange(len(target_ids_shifted))[mask_flat]
+                valid_targets = target_ids_shifted[mask_flat]
+            if DEBUG:
+                print(f"[VALID_INDICES] Valid indices shape: {valid_indices.shape}")
+                print(f"[VALID_TARGETS] Valid targets shape: {valid_targets.shape}")
+            # Gather the log probabilities for the correct target tokens
+            valid_log_probs = pred_log_probs[valid_indices, valid_targets]
```
[the remark this answers is not in the document]
Comment (ynankani) [reply examples/windows/accuracy_benchmark/perplexity_metrics/perplexity_metrics.py:519]: as part of prefill chunking enabled logits returned are for the last chunk_seq_len so trg_len is same as chunk_seq_len.

**Comment by vishalpandya1990** [comment]
In KL-Divergence, Is the single-model script (hf/onnx comparison) exact subset of unified script? Do we need to publish both?

Overall looks okay to me. Please get the scripts reviewed from Anuj Jalota as well. And, skim through the coderabbit review comments.

**Comment by ynankani** [comment]
the unified script logic is similar to the single ep script, but the single ep script runs for the whole dataset while the unified script runs for some chunks of dataset due to memory constraint so kept both, the unified script delegates to same ep script when comparing two model with same ep.
> In KL-Divergence, Is the single-model script (hf/onnx comparison) exact subset of unified script? Do we need to publish both?
> 
> Overall looks okay to me. Please get the scripts reviewed from Anuj Jalota as well. And, skim through the coderabbit review comments.

**Comment by anujj** [comment]
Possible to add the assert in the code that for the chunking case .. chunk_size and the strid will always be same ?

**Review by i-riyad — APPROVED** [review]
